Refactor code

iamargus95 opened the pull request:
1. Changed githubapi.UserData and githubapi.RepoData to struct methods.
2. Added Functionality to take multiple usernames from CLI. 
3. Added concurrency
4. Added makefile and build instructions in README.md

iamargus95 commented:
Added ability to parse usernames directly from the CLI.

jdk2588 commented on githubapi/githubapi.go:
```diff
@@ -8,6 +8,8 @@ import (
 	"strconv"
 )
 
+const UrlGetRequest = "https://api.github.com/users/"
```
Constant are in CAPS https://golang.org/doc/effective_go#constants

jdk2588 commented on githubapi/githubapi.go:
```diff
@@ -41,7 +43,7 @@ func responseToRepoData(data []byte) ReposInfoArray {
 func GetUserData(username string) Userinfo {
 
-	url := "https://api.github.com/users/" + username
+	url := UrlGetRequest + username
```
Please use url constructor library, like https://pkg.go.dev/net/url

jdk2588 commented on githubapi/githubapi.go:
```diff
@@ -71,7 +73,7 @@ func GetReposData(username string, noOfRepos int) ReposInfoArray {
 	for i := 1; i <= ((noOfRepos / 100) + 1); i++ {
 
-		url := "http://api.github.com/users/" + username + "/repos?type=public&per_page=100&page=" + strconv.Itoa(i)
+		url := UrlGetRequest + username + "/repos?type=public&per_page=100&page=" + strconv.Itoa(i)
```
Please use url constructor library, like https://pkg.go.dev/net/url

jdk2588 commented on main.go:
```diff
@@ -1,26 +1,36 @@
+	for _, username := range usernames {
+		wg.Add(1)
+		go worker(username, dataToFile, &wg)
+		io.WriteToFile(username, dataToFile)
```
Why write to file is outside of Go routine?

jdk2588 commented on io/writeToFile.go:
```diff
@@ -6,7 +6,9 @@ import (
 )
 
-func WriteToFile(username string, data []string) {
+func WriteToFile(username string, dataToFile chan []string) error {
```
Write to File should only care about content like string data, should not use any channel.

iamargus95 commented:
Fixed Concurrency issues where data was being written to random files depending on which api call gives a faster response. Workaround :- Used a channel of map[string][]string to overcome it.

Fixed waitgroups not working as expected.  [ defer wg.Done() would not work inside the fetch worker func. ] 
Workaround :- Did a wg.Add(len(usernames)) before the write worker func.

Added Benckmarking tests and updated the README.
